File: services/wordpress.py
import os
from base64 import b64encode

import requests


class WordPressNotConfigured(Exception):
    """Raised when WP env vars are missing."""


def publish_to_wordpress(title: str, content: str, status: str | None = None) -> str:
    url = os.getenv("WP_URL")
    user = os.getenv("WP_USER")
    app_pwd = os.getenv("WP_APP_PASSWORD")
    if not (url and user and app_pwd):
        raise WordPressNotConfigured()

    status = status or os.getenv("WP_DEFAULT_STATUS", "draft")
    api = url.rstrip("/") + "/wp-json/wp/v2/posts"
    auth = b64encode(f"{user}:{app_pwd}".encode()).decode()

    r = requests.post(
        api,
        headers={
            "Authorization": f"Basic {auth}",
            "Content-Type": "application/json",
        },
        json={"title": title, "content": content, "status": status},
        timeout=30,
    )
    r.raise_for_status()
    data = r.json()
    return data.get("link") or data.get("guid", {}).get("rendered", "")

File: src/lib/services/wordpress.ts
export class WordPressNotConfigured extends Error {
  constructor() {
    super('WordPress credentials not configured')
  }
}

export async function publishToWordPress(
  title: string,
  content: string,
  status = process.env.WP_DEFAULT_STATUS ?? 'draft',
): Promise<string> {
  const url = process.env.WP_URL
  const user = process.env.WP_USER
  const password = process.env.WP_APP_PASSWORD

  if (!url || !user || !password) throw new WordPressNotConfigured()

  const credentials = Buffer.from(`${user}:${password}`).toString('base64')

  const response = await fetch(`${url}/wp-json/wp/v2/posts`, {
    method: 'POST',
    headers: {
      Authorization: `Basic ${credentials}`,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ title, content, status }),
  })

  if (!response.ok) {
    const error = await response.text()
    throw new Error(`WordPress API error ${response.status}: ${error}`)
  }

  const post = (await response.json()) as { link: string }
  return post.link
}
